test/unit/org/apache/cassandra/index/sai/metrics/QueryMetricsTest.java

```diff
@@ -108,6 +111,60 @@ public void testMetricRelease() throws Throwable
         assertThatThrownBy(() -> getTableQueryMetrics(keyspace, table, "TotalQueriesCompleted")).hasCauseInstanceOf(InstanceNotFoundException.class);
     }
 
+    @Test
+    public void testIndexQueryWithPartitionKey() throws Throwable
+    {
+        String table = "test_range_key_type_with_index";
+        String index = "test_range_key_type_with_index_index";
+
+        String keyspace = createKeyspace(CREATE_KEYSPACE_TEMPLATE);
+
+        createTable(String.format(CREATE_TABLE_TEMPLATE, keyspace, table));
+        createIndex(String.format(CREATE_INDEX_TEMPLATE, index, keyspace, table, "v1"));
+
+        int rowsWrittenPerSSTable = 10;
+        int numberOfSSTable = 5;
+        int rowsWritten = 0;
+        int i = 0;
+        for (int j = 0; j < numberOfSSTable; j++)
+        {
+            rowsWritten += rowsWrittenPerSSTable;
+            for (; i < rowsWritten; i++)
+            {
+                execute("INSERT INTO " + keyspace + "." + table + " (id1, v1, v2) VALUES (?, ?, '0')", Integer.toString(i), i);
+            }
+            flush(keyspace, table);
+        }
+
+        waitForIndexQueryable(keyspace, table);
+
+//        ResultSet rows1 = executeNet("SELECT id1 FROM " + keyspace + "." + table + " WHERE v1 = 3");
```

Comment:
is it needed?

Reply:
I think we might need this since the queries are related to indexes.